pkgs/dart_tooling_mcp_server/test/test_harness.dart

```diff
@@ -47,40 +48,34 @@ class TestHarness {
   /// breakpoints in the server don't work however, so you can set [debugMode]
   /// to `true` and we will run it in process instead, which allows breakpoints
   /// since everything is running in the same isolate.
+  ///
+  /// Use [startDebugSession] to start up apps and connect to them.
   static Future<TestHarness> start({
     @Deprecated('For debugging only, do not submit') bool debugMode = false,
   }) async {
-    final platform = Platform.isLinux
-        ? 'linux'
-        : Platform.isMacOS
-            ? 'macos'
-            : throw StateError(
-                'unsupported platform, only mac and linux are supported',
-              );
-    final flutterProcess = await TestProcess.start(
-      // TODO: Get flutter SDK location from somewhere.
-      'flutter',
-      ['run', '-d', platform, '--print-dtd'],
-      workingDirectory: counterAppPath,
-    );
-    addTearDown(() async {
-      flutterProcess.stdin.writeln('q');
-      await flutterProcess.shouldExit(0);
-    });
-
-    final dtdUri = await _getDTDUri(flutterProcess);
-
     final mcpClient = DartToolingMCPClient();
     addTearDown(mcpClient.shutdown);
     final connection = await _initializeMCPServer(mcpClient, debugMode);
 
-    final fakeEditorExtension = await FakeEditorExtension.connect(
-      flutterProcess,
-      dtdUri,
-    );
+    final dtdProcess = await TestProcess.start('dart', ['tooling-daemon']);
+    final dtdUri = await _getDTDUri(dtdProcess);
+
+    final fakeEditorExtension = await FakeEditorExtension.connect(dtdUri);
```

Comment:
technically to mimic the IDE environment, the fake editor should start DTD and pass the workspace roots (this is a protected operation that only the IDE, or the thing that starts DTD, can do). Idk if this difference matters right now or not, but if workspace roots need to be set on DTD for anything, it may.

Reply:
It probably doesn't ultimately matter right now - is that protected by the extra "client secret" that I saw it log?

Reply:
Fwiw MCP also has a notion of project roots, so right now the MCP server is asking the client for that not DTD

Reply:
> is that protected by the extra "client secret"

Yeah. So only the IDE who has the key can set these.

Reply:
Yeah ok, so we do have access to it and could plumb it through if we need to